Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp

```diff
@@ -0,0 +1,255 @@
+        auto sb1 = decoder.GetSoftwareBitmapAsync().get();
+        sbList.Append(SoftwareBitmap::Convert(sb1, BitmapPixelFormat::Nv12));
+    }
+    // photo2
+    {
+        auto strm = file2.OpenReadAsync().get();
+        auto decoder = BitmapDecoder::CreateAsync(strm).get();
+        auto sb1 = decoder.GetSoftwareBitmapAsync().get();
+        sbList.Append(SoftwareBitmap::Convert(sb1, BitmapPixelFormat::Nv12));
+    }
+
+    // Do some processing like low light fusion in this example
+    auto op = LowLightFusion::FuseAsync(sbList).get();
+
+    // Save the output
+    auto fname = to_hstring(photoIndex) + L"_LLF.png";
+    auto fileOp = folder.CreateFileAsync(fname, CreationCollisionOption::ReplaceExisting).get();
+    {
+        auto strm = fileOp.OpenAsync(FileAccessMode::ReadWrite).get();
+        auto encoder = BitmapEncoder::CreateAsync(BitmapEncoder::PngEncoderId(), strm).get();
+        encoder.SetSoftwareBitmap(SoftwareBitmap::Convert(op.Frame(), BitmapPixelFormat::Bgra8));
+        encoder.FlushAsync().get();
+    }
+}
+
+MediaCapture InitCamera()
+{
+    MediaCapture mediaCapture;
+    auto settings = MediaCaptureInitializationSettings();
+    auto filteredGroups = GetFilteredSourceGroupList();
+
+    if (filteredGroups.Size() < 1)
+    {
+        std::cout << "Error No suitable capture sources found";
+        throw_hresult(MF_E_NO_CAPTURE_DEVICES_AVAILABLE);
+    }
+
+    auto selectedSGidx = GetSGSelection(filteredGroups);
+
+    auto selectedSrc = filteredGroups.GetAt(selectedSGidx);
+
+    if (selectedSrc == nullptr)
+    {
+        throw_hresult(MF_E_OUT_OF_RANGE);
+    }
+
+    settings.SourceGroup(selectedSrc.SourceGroup());
+
+    // if user explicitly selected a non-photo pin to take the photo
+    if (selectedSrc.MediaStreamType() != MediaStreamType::Photo)
+    {
+        settings.PhotoCaptureSource(PhotoCaptureSource::VideoPreview);
+    }
+    else
+    {
+        // we hope that auto will select the best photo pin options
+        settings.PhotoCaptureSource(PhotoCaptureSource::Auto);
+    }
+
+    settings.StreamingCaptureMode(StreamingCaptureMode::Video);
+    mediaCapture.InitializeAsync(settings).get();
+
+    //Set format on the medicapture frame source
+    auto formatIdx = GetMediaTypeSelection(selectedSrc);
+    auto frameSourceIter = mediaCapture.FrameSources().First();
+    while (frameSourceIter.HasCurrent())
+    {
+
+        auto frameSource = frameSourceIter.Current().Value();
+        if (frameSource.Info().Id() == selectedSrc.Id())
+        {
+            frameSource.SetFormatAsync(frameSource.SupportedFormats().GetAt(formatIdx)).get();
+        }
+        frameSourceIter.MoveNext();
+    }
+    return mediaCapture;
+}
+
+int wmain()
+{
+    std::cout << "Windows Media Capture in Win32 desktop Console app!";
+    auto mediaCapture = InitCamera();
+
+    // Main processing part of photos
+    std::cout << "Press 'p' take photos and 'q' to quit";

```

Comment:
>std::cout << "Press 'p' take photos and 'q' to quit"; [](start = 4, length = 53)

suggestion: Move this in your while loop and add some characters to format this: i.e. std::cout << "-----" << "Do whatever" << "-----" << std:endl;